homework/src/index.js

```diff
@@ -1,19 +1,31 @@
 'use strict';
 
 {
-  function fetchJSON(url, cb) {
-    const xhr = new XMLHttpRequest();
-    xhr.open('GET', url);
-    xhr.responseType = 'json';
-    xhr.onload = () => {
-      if (xhr.status < 400) {
-        cb(null, xhr.response);
-      } else {
-        cb(new Error(`Network error: ${xhr.status} - ${xhr.statusText}`));
-      }
-    };
-    xhr.onerror = () => cb(new Error('Network request failed'));
-    xhr.send();
+  function fetchJSON(url, container) {
+    return new Promise((resolve, reject) => {
+      const xhr = new XMLHttpRequest();
+
+      const loadingGif = createAndAppend('img', container, {
+        src: './P1Chi6u1.gif',
+        alt: 'loading gif',
+        class: 'loadingGif'
+      });
+      xhr.onreadystatechange = () => {
```

Comment:
This looks like a nice gimmick but really only makes sense if, on average, you expect a response to take considerable time. In this particular case the response comes so quickly (< 200ms) that the GIF only produces an incomprehensible flicker. So definitely use this in other cases where the response takes time, but in this case I would leave it out.

Also, as a general principle, you should try and avoid mixing 'concerns' in a single function. The `fetchJSON()` that deals with fetching data should not also be concerned with manipulating the DOM.